examples/Cloud_Robotics/singletask_learning_bench/Semantic_Segmentation/testalgorithms/basemofel.py

```diff
@@ -0,0 +1,175 @@
+
+        for epoch in range(self.trainer.args.start_epoch, self.trainer.args.epochs):
+            if epoch == 0 and self.trainer.val_loader:
+                self.trainer.validation(epoch)
+            self.trainer.training(epoch)
+
+            if self.trainer.args.no_val and \
+                    (epoch % self.trainer.args.eval_interval == (self.trainer.args.eval_interval - 1)
+                     or epoch == self.trainer.args.epochs - 1):
+                # save checkpoint when it meets eval_interval or the training finished
+                is_best = False
+                self.checkpoint_path = self.trainer.saver.save_checkpoint({
+                    'epoch': epoch + 1,
+                    'state_dict': self.trainer.model.state_dict(),
+                    'optimizer': self.trainer.optimizer.state_dict(),
+                    'best_pred': self.trainer.best_pred,
+                }, is_best)
+
+            # if not self.trainer.args.no_val and \
+            #         epoch % self.train_args.eval_interval == (self.train_args.eval_interval - 1) \
+            #         and self.trainer.val_loader:
+            #     self.trainer.validation(epoch)
+            # self.checkpoint_path = os.path.join(self.temp_dir, '{}_'.format(
+            #                 cfgs.DATASET_NAME) + str(_global_step) + "_" + str(time.time()) + '_model.ckpt')
+            #             saver.save(sess, self.checkpoint_path)
+            #             print('Weights have been saved to {}.'.format(self.checkpoint_path))
+
+        self.trainer.writer.close()
+
+        return self.checkpoint_path
+
+    def save(self, model_path):
+        if not model_path:
+            raise Exception("model path is None.")
+
+        model_dir, model_name = os.path.split(self.checkpoint_path)
+        models = [model for model in os.listdir(
+            model_dir) if model_name in model]
+
+        if os.path.splitext(model_path)[-1] != ".zip":
+            model_path = os.path.join(model_path, "model.zip")
+
+        if not os.path.isdir(os.path.dirname(model_path)):
+            os.makedirs(os.path.dirname(model_path))
+
+        with zipfile.ZipFile(model_path, "w") as f:
+            for model_file in models:
+                model_file_path = os.path.join(model_dir, model_file)
+                f.write(model_file_path, model_file,
+                        compress_type=zipfile.ZIP_DEFLATED)
+
+        return model_path
+
+    def predict(self, data, **kwargs):
+        if not isinstance(data[0][0], dict):
+            data = self._preprocess(data)
+
+        if type(data) is np.ndarray:
+            data = data.tolist()
+
+        self.validator.test_loader = DataLoader(
+            data, batch_size=self.validator.args.test_batch_size, shuffle=False, pin_memory=True)
+        return self.validator.validate()
+
+    def load(self, model_url):
+        model_url = '/home/wxc/dev/ianvs/models/model_best_mapi_only.pth'
```

Comment:
The model_url is hardcoded in the load function which would bypass the parameter, which should be revised.